Makefile

```diff
@@ -136,7 +136,10 @@ docker-cp-coverage:
 web:
 	($(VENV_RUN); bin/localstack web)
 
-test:              ## Run automated tests
+## Run automated tests
+test:
+	# elasticmq is required as SQS backend for X-Ray testing
+	export SQS_PROVIDER=elasticmq; \
```

Comment:
Wouldn't this trigger the test only for elasticmq and not for (moto) sqs? 
We would be replacing every moto test with this but at the same time moto is the default runner for SQS.

I was thinking of creating another separate integration test only for elasticmq, but at the same time the infra starter does not allow us to create two different listeners nor putting one service down to restart it as elasticmq, and duplicating the tests just to do it with elasticmq would take a lot more time.

Any thoughts?

Reply:
This also fixes https://github.com/localstack/localstack/issues/3528, can you confirm @kubawach?

Reply:
@mgagliardo yes it does, I needed SQS XRay propagation support for this so re-used your code from that PR :) Thanks by the way!

Reply:
Regarding the change in Makefile, you're completely right. As far as I understand the build, previously all SQS tests were running on moto only. Having two tests run per SQS implementation (with some split between which runs which due to incompatibilities) would be definitely the right approach. I'll take a look into this.

Reply:
Hi @kubawach . Sorry for the long delay on this one!

Instead of extending the `test` target here, I think we should rather add a new line to the `scripts` section in `.travis.yml` that would run only the new test method with the elasticmq provider:
```
- SQS_PROVIDER=elasticmq TEST_PATH=tests/integration/test_sns.py:SNSTest.test_publish_sqs_from_sns_with_xray_propagation make test
```
Let's add the test after line 51 here: https://github.com/localstack/localstack/blob/master/.travis.yml#L51

Reply:
Done.